test-resources/bin/lib/util

```diff
@@ -747,25 +747,29 @@ build_test_application() {
     return
   fi
 
-  local app_path="$1"
-  local image_name="$2"
-
-  pushd "$app_path" > /dev/null
-  echo "building container image for application under monitoring ($image_name)"
-  docker build . -t "$image_name"
-  popd > /dev/null
+  local make_rule="$1"
+  TEST_APP_DOTNET_IMAGE_REPOSITORY="$test_app_dotnet_image_repository" \
+    TEST_APP_DOTNET_IMAGE_TAG="$test_app_dotnet_image_tag" \
+    TEST_APP_JVM_IMAGE_REPOSITORY="$test_app_jvm_image_repository" \
+    TEST_APP_JVM_IMAGE_TAG="$test_app_jvm_image_tag" \
+    TEST_APP_NODEJS_IMAGE_REPOSITORY="$test_app_nodejs_image_repository" \
+    TEST_APP_NODEJS_IMAGE_TAG="$test_app_nodejs_image_tag" \
```

Comment:
The need to list all these env vars whenever calling make is quite ugly, maybe we can get rid of it later. For now I think it is an acceptable interim solution.

I think using the make rules instead of a direct `docker build` / `docker push` might be slightly preferable in terms of consistency. Maybe debatable.